### ISSUE TITLE
IntelliSense with different file system types (Windows 10)

### ISSUE THREAD

**Issue by jordangarside** [microsoft/vscode-cpptools#1046]
- VSCode Version: Code - Insiders 1.17.0-insider (128a4e3640c5375f7b0bc1cfd37f626481b1386f, 2017-09-12T05:24:19.607Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
cpptools|ms-|0.12.4

IntelliSense doesn't seem to be working with SSHFS filesystem type in Windows 10 (no import errors detected and no squiggly lines). I know this is a fringe case, but I'm wondering if there is an easy fix.

IntelliSense works fine for this example in the local filesystem as well as on an exfat formatted flash drive.

---

Steps to Reproduce:

1. Install winsshfs and mount a remote file system
2. Open the a project locally from that mounted file system

<img width="270" alt="nlogn_properties" src="https://user-images.githubusercontent.com/6321824/30560600-715fce7c-9c86-11e7-9f4e-7f84be4a01e0.png">
<img width="761" alt="helloworld_vscode" src="https://user-images.githubusercontent.com/6321824/30560601-7163b1ea-9c86-11e7-9996-ffeeb6b7298e.png">

Reproduces without extensions: No (Need the C++ extension for intellisense)

**Comment by jophippe** [comment]
According to #735, it looks like sshfs mounts that are too large will fail to be searched. You could try using a smaller mount and see if that works

**Comment by jordangarside** [comment]
Gotcha, thanks a lot @jophippe 
I should have searched for that first, my bad.

**Comment by sean-mcmanus** [comment]
@jordangarside https://github.com/Microsoft/vscode-cpptools/issues/735 was with Mac and the file APIs are different. Did using a smaller mount fix the issue?

**Comment by jordangarside** [comment]
I'm going to try a smaller mount today @sean-mcmanus . I'll post here with the results.

**Comment by jordangarside** [comment]
@sean-mcmanus Just tried a 50GB mount and still had the same issue. Strangely, it looks like IntelliSense is still loading because I get the `Configure includePath for better IntelliSense results.` message (this happens for both mount sizes), but it doesn't detect the import errors.

There are also no problems reported in the console.

**Comment by sean-mcmanus** [comment]
Which winsshfs are you using? This one: https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/win-sshfs/win-sshfs-0.0.1.5-setup.exe ?

**Comment by jordangarside** [comment]
@sean-mcmanus [https://github.com/Foreveryone-cz/win-sshfs/releases/tag/1.6.1](https://github.com/Foreveryone-cz/win-sshfs/releases/tag/1.6.1)

**Comment by sean-mcmanus** [comment]
https://github.com/Foreveryone-cz/win-sshfs/releases/download/1.6.1/WinSSHFS-1.6.1.13-devel.msi doesn't install on any of my machines. Gives a couple errors: 2503, 2502. Are you using something else to install it?

**Comment by jordangarside** [comment]
I'm using that MSI to install.

<img width="438" alt="winsshfs" src="https://user-images.githubusercontent.com/6321824/30613490-4b08884c-9d56-11e7-8848-818e5fe82352.png">

I think you also need to have [Dokany](https://github.com/dokan-dev/dokany/releases/tag/1.0.5) installed before running the MSI.

**Comment by sean-mcmanus** [comment]
Yeah, I have 1.0.5.1000 of Dokan: https://github.com/dokan-dev/dokany/releases/download/1.0.5/DokanSetup-1.0.5.1000.exe .

**Comment by jophippe** [comment]
Okay, I just set up an sshfs connection using the 4every1 edition 1.5.12.8 and Dokany 0.7.4. I set up a folder on the mount, added an include, and it didn't show an error, just like yours. When I hover over the include, I get the message "FE 'Compiler exited with error - No IL available'". Do you get the same message @jordangarside?

**Comment by jordangarside** [comment]
Yup I get the same message @jophippe.

**Comment by jophippe** [comment]
Okay. The compiler failing is almost certainly the cause of the missing squiggles. I'll throw a debugger on there and see if this is a simple fix. Thanks for reporting this issue @jordangarside

**Comment by jordangarside** [comment]
Great, thanks guys!

**Comment by barberdt** [comment]
Not sure if it's 100% related, but trying to get some feedback regarding the issues I've mentioned here https://github.com/Microsoft/vscode/issues/13665. I'm using VSCode, latest build (insiders has the same issue as well) on MacOS with a 1.2G SSHFS mounted directory. Intellisense and autocomplete are completely non-existent. FWIW, I'm mostly editing JS. It's quite frustrating given how much I love the editor (and that Atom won't even handle opening the directory), but having no autocomplete is...painful. Happy to provide any more info that's relevant. This should likely be an issue for anyone else who has to work on a monorepo hosted in a dev VM using SSHFS such that the files don't stay on the local machine.

**Comment by garyng** [comment]
Is there any updates for this?

I got the same error while hovering over the includes:

![image](https://user-images.githubusercontent.com/4453664/57349559-311fb600-718d-11e9-9b95-43d0affdc650.png)

Edit: the error seems to be similar to https://github.com/microsoft/vscode-cpptools/issues/1694

```
Unable to retrieve file system information for xxx. error = -1
```

**Comment by kasvith** [comment]
Having the same issue

**Comment by Michanne** [comment]
I get this issue when trying to add a WSL-created directory as an include path. Create any directory from WSL and try to use it as an include path, giving the error:

`Unable to retrieve file system information for xxx. error = -1`

The directory works fine, is not corrupted, files can be accessed as normal from the file explorer as well as VSCode, but completely breaks the Intellisense apparently. Makes it impossible to use for anything related to WSL

**Comment by sean-mcmanus** [comment]
@Michanne Are you using WSL2? Are you using the Remote - WSL extension to the connect? That is the recommended method.

**Comment by Michannne** [comment]
@sean-mcmanus No not using WSL2, and I tried both with the Remote extension and also just creating a fresh project and adding a WSL folder to the browse.path or includePath depending on the engine.

I was able to resolve this issue another way, I create a folder for my repos by symlinking a C:/ drive folder, i.e. ~/repos -> /mnt/c/repos. 

Like this, I am able to use intellisense on the Windows-side. Otherwise, if the directory is located in the rootfs/ of the WSL installation, it won't work, regardless of permissions. Intellisense can't read any directory in rootfs/

**Comment by pobinger** [comment]
Not an answer, but maybe helpful for anyone else.
I had a similar problem with Visual Studio Code and its Intellisense C++ extension.
It didn't work because (on Windows 10) you can use 'fsutil' to change directories to be case sensitive.

Don't do that to SysGCC's directory and the checkout to 'case sensitive', it prevents the VS code plugin from successfully finding the source files.

**Comment by sean-mcmanus** [comment]
We have a C_Cpp.caseSensitiveFileSupport setting you can change to "enabled" if you need to use case sensitive directories/files on Windows (Linux/Mac are already case sensitive by default).
